tests/sentry/seer/code_review/test_webhooks.py

```diff
@@ -0,0 +1,155 @@
+        """Test that missing external_id is handled gracefully."""
+        # Create event without external_id
+        event_without_external_id = orjson.loads(CHECK_RUN_REREQUESTED_ACTION_EVENT_EXAMPLE)
+        del event_without_external_id["check_run"]["external_id"]
+
+        with self.options({"coding_workflows.code_review.github.check_run.rerun.enabled": True}):
+            with patch("sentry.seer.code_review.webhooks.logger") as mock_logger:
+                self._send_check_run_event(orjson.dumps(event_without_external_id))
+
+                # Verify NO request was made to Seer
+                mock_request.assert_not_called()
+                # Verify error was logged (validation errors are logged with exception)
+                mock_logger.exception.assert_called_once()
+                assert (
+                    "Invalid GitHub check_run event payload"
+                    in mock_logger.exception.call_args[0][0]
+                )
+
+    @patch("sentry.seer.code_review.webhooks.make_signed_seer_api_request")
+    @with_feature({"organizations:gen-ai-features", "organizations:seat-based-seer-enabled"})
+    def test_check_run_fails_when_external_id_not_numeric(self, mock_request: MagicMock) -> None:
+        """Test that non-numeric external_id is handled gracefully."""
+        # Create event with non-numeric external_id
+        event_with_invalid_external_id = orjson.loads(CHECK_RUN_REREQUESTED_ACTION_EVENT_EXAMPLE)
+        event_with_invalid_external_id["check_run"]["external_id"] = "not-a-number"
+
+        with self.options({"coding_workflows.code_review.github.check_run.rerun.enabled": True}):
+            with patch("sentry.seer.code_review.webhooks.logger") as mock_logger:
+                self._send_check_run_event(orjson.dumps(event_with_invalid_external_id))
+
+                # Verify NO request was made to Seer
+                mock_request.assert_not_called()
+                # Verify error was logged (ValueError for non-numeric external_id)
+                mock_logger.exception.assert_called_once()
+                assert "external_id must be numeric" in mock_logger.exception.call_args[0][0]
+
+    @patch("sentry.seer.code_review.webhooks.make_signed_seer_api_request")
+    @with_feature({"organizations:gen-ai-features", "organizations:seat-based-seer-enabled"})
+    def test_check_run_handles_seer_error_gracefully(self, mock_request: MagicMock) -> None:
+        """Test that Seer errors are caught and logged without failing the webhook."""
+        # Mock Seer API to return an error
+        mock_request.return_value = self._mock_response(500, b'{"error": "Internal server error"}')
+
+        with self.options({"coding_workflows.code_review.github.check_run.rerun.enabled": True}):
+            with patch("sentry.seer.code_review.webhooks.logger") as mock_logger:
+                self._send_check_run_event(CHECK_RUN_REREQUESTED_ACTION_EVENT_EXAMPLE)
```

Comment:
Even though Seer is returning 500, we will assert that status code is 204, thus, GitHub will not retry.